additionalResolvers/quiz.ts

```diff
@@ -0,0 +1,1820 @@
+                                    }
+                             }
+                            
+                    }
+                    `,
+                    context,
+                    info
+                });
+                let assessment =assessments[0];
+                let oldItems= assessment.items;
+                let items=assessment.items;
+                items=[...items,_args.item];
+                assessment.items=items;
+                let selectionSet=`{updateAssessment(input: {
+                    metadata: {
+                        name: "${assessment.metadata.name}",
+                        suggestedDate: "${assessment.metadata.suggestedDate}",
+                        chapterId: "${assessment.metadata.chapterId}",
+                        rewardPoints: ${assessment.metadata.rewardPoints},
+                        tagNames: [${assessment.metadata.tagNames.map(tag => `"${tag}"`)}]
+                    },
+                    assessmentMetadata: {
+                        skillPoints: ${assessment.assessmentMetadata.skillPoints},
+                        skillTypes: [${assessment.assessmentMetadata.skillTypes.map(skillType => `${skillType}`)}],
+                        initialLearningInterval: ${assessment.assessmentMetadata.initialLearningInterval}
+                    },
+                    items:[
+                        ${assessment.items.map(item => `{
+                            ${item.id != null ? `id:"${item.id}",` : ''}
+                            associatedSkills:[${item.associatedSkills.map(skill => `{ ${skill.id != null ? `id:"${skill.id}",` : ''} skillName:"${skill.skillName}"}`)}],
+                            associatedBloomLevels:[${item.associatedBloomLevels.map(level => `${level}`)}]
+                        }`)}
+                    ]
+                }) {
+                    id
+                    items{
+                        id
+                        associatedSkills{
+                            id
+                            skillName
+                        }
+                        associatedBloomLevels
+                    }
+                }
+                }`;
+                let updatedItems = await context.ContentService.Mutation.mutateContent({
+                    root,
+                    args:{contentId:_args.assessmentId},
+                    selectionSet:selectionSet, 
+                    context,
+                    info});
+
+                let questionInput=_args.questionInput;
+                for (let item of updatedItems.updateAssessment.items) {
+                    if (!oldItems.some(oldItem => oldItem.id === item.id)) {
+                        questionInput.itemId= item.id;
+                        break;
+                    }
+                }
+                let selectionSetQuiz=`{_internal_noauth_addMultipleChoiceQuestion(input: {
+                    itemId: "${questionInput.itemId}",
```

Comment:
Why does it expect an ID to be passed for the question? The question doesn't exist yet, so there is no ID and shouldn't the `_internal_noauth_addMultipleChoiceQuestion` mutation itself create a new UUID for the question? (This would however make it necessary that the code which updates the skill metadata of this assessment in the content service is called AFTER creating the question. Would that be a problem?) 

Firstly creating the item itself and get its id and then, afterwards, updating the metadata in the content service, would make more sense IMO and would also be consistent with how other parts of the system work, e.g. flashcards and quizzes themselves are first created and then their ID passed to the content service. Why flip this for the assessment items?